filebeat/docs/inputs/input-filestream-file-options.asciidoc

```diff
@@ -14,6 +14,19 @@ The prospector is running a file system watcher which looks for files specified
 in the `paths` option. At the moment only simple file system scanning is
 supported.
 
+[float]
+[[filestream-input-id]]
+===== `id`
+
+A unique identifier for this filestream input. Each filestream input
+must have a unique ID.
+
+WARNING: If there are multiple inputs without an ID, there will be
+data duplication when {beatname_lc} is restarted.
+
+WARNING: Adding an ID to an existing configuration will read the files
+from the beginning.
```

Comment:
This warning and the previous one together are telling users that they need an ID to avoid data duplication but also that adding a ID will cause data duplication.

It reads like data duplication is unavoidable eventually here, is this actually the case? Is there anything we can do to improve this?

Reply:
As of now this is the case: if IDs are added, files will be re-read.

However in the near future (aka my next task) we will have a migration path to handle adding IDs for the first time.

My main goal with those warnings is the following message:
- Adding ID to an existing input will duplicate data.
- New inputs must only be created with IDs.
- New users should only create inputs with IDs.

I just haven't found a nice way to write that. Also this is a temporary situation, once IDs are required, then there will be no issues and those warnings can be removed.